File: app/src/main/java/com/antonio/samir/meteoritelandingsspots/service/server/nasa/NasaServiceFactory.java
package com.antonio.samir.meteoritelandingsspots.service.server.nasa;

public class NasaServiceFactory {
    private static NasaService nasaService;

    public static NasaService getNasaService() {

        final NasaService service;

        if (nasaService != null) {
            service = nasaService;
        } else {
            service = new NasaServiceImpl();
        }

        return service;
    }
}

File: app/src/main/java/com/antonio/samir/meteoritelandingsspots/service/server/nasa/NasaServiceFactory.kt
package com.antonio.samir.meteoritelandingsspots.service.server.nasa

object NasaServiceFactory {
    private val nasaService: NasaService? = null

    fun getNasaService(): NasaService {

        val service: NasaService

        if (nasaService != null) {
            service = nasaService
        } else {
            service = NasaServiceImpl()
        }

        return service
    }
}
